File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task109_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter your email:");
        String email = scanner.nextLine();
        String regex = "^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(email);
        if (matcher.matches()) {
            System.out.println("Email is valid");
        } else {
            System.out.println("Email is not valid");
        }
    }
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task109_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <regex>
#include <string>

int main() {
    std::string email;
    std::cout << "Enter your email:";
    std::getline(std::cin, email);
    std::regex regex("^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$");
    if (std::regex_match(email, regex)) {
        std::cout << "Email is valid" << std::endl;
    } else {
        std::cout << "Email is not valid" << std::endl;
    }
    return 0;
}